src/puppetlabs/puppetdb/command/dlo.clj

```diff
@@ -71,31 +76,43 @@
                           (.printStackTrace exception out))
                         attempts))))
 
+(defn- strip-metadata-suffix
+  [queue-metadata-str]
+  (str/replace queue-metadata-str #"\.json$" ""))
```

Comment:
Hmm, well if we need to be worried about it not having the correct suffx, then I'd wonder if we might want to verify that in either approach, but I was originally just thinking about it from the perspective that subs is probably a lot simpler.

I suppose if you wanted to use subs, and if a missing extension was ok, you could guard it with an "if endswith?", but in any case subs vs re isn't a big deal to me either way, since the strings should be fairly short, I just mentioned it offhand.  (With subs, I suspected that the implementation might be more likely to be able to "work from the right" (plus or minus some utf-16 sync issues).  If re can't too, then it might be O(n) vs O(1).)